icaruscode/Decode/DecoderTools/TriggerDecoderV3_tool.cc

```diff
@@ -295,14 +320,20 @@ namespace daq
       { return static_cast<long long int>(a) - static_cast<long long int>(b); }
 
     /// Fills one `cryostat` information of `sbn::ExtraTriggerInfo`.
-    static sbn::ExtraTriggerInfo::CryostatInfo unpackPrimitiveBits(
+    sbn::ExtraTriggerInfo::CryostatInfo unpackPrimitiveBits(
       std::size_t cryostat, bool firstEvent, unsigned long int counts,
       std::uint64_t connectors01, std::uint64_t connectors23
-      );
+      ) const;
 
+    /// Encodes all the `connectorWord` LVDS bits for the specified `cryostat`.
+    std::array<std::uint64_t, sbn::ExtraTriggerInfo::MaxWalls> encodeLVDSbits(
+      short int cryostat,
+      std::uint64_t connector01word, std::uint64_t connector23word
+      ) const;
+    
     /// Encodes the `connectorWord` LVDS bits from the specified `cryostat`
     /// and `connector` into the format required by `sbn::ExtraTriggerInfo`.
-    static std::uint64_t encodeLVDSbits
+    static std::uint64_t encodeLVDSbitsLegacy
```

Comment:
It is a fallback in case the channel mapping database is not available.
I thought of preserving _a bit of_ the functionality in a possible context where database access is problematic (note however that by default `AllowDefaultLVDSmap` is `false`, and the channel mapping service is required).
The "legacy" mapping of the LVDS is "almost right" in the sense that typically the bit is close to where it should be, but it may be not exactly there.

The other case where legacy mapping kicks in is when the database lacks the proper information. Effectively this is a non-issue, since @mvicenzi has updated all run periods with the data needed for the PMT pair mapping, and while the adders information is present only for Run3, but so are the adder bits, so before then there is nothing to decode.

* A bit is associated to a defined location according to the prescription in `sbn::ExtraTriggerInfo`, which says the most upstream PMT end in the most significant bits. This prescription is not fully... prescriptive though: consider for example that the first two bits include PMT channels #0 and #1 respectively, and these two PMT have the same _z_ coordinate.

Reply:
The choice of whether the legacy mapping is used or not is automatic -- the only direct control the user has is whether to ever allow it to kick in (`AllowDefaultLVDSmap: true`) or not (the default).
If allowed, the legacy kicks in if the mapping database is not available (which will be a fatal issue for other decoders, I think). For example, one could think that in the online monitor environment database access is preferably avoided.
Users should not want the legacy mapping for any reason if there is choice.